actions/trigger-argo-workflow/action.yaml

```diff
@@ -0,0 +1,136 @@
+name: Trigger Argo Workflow
+description: Trigger an Argo workflow in the Grafana Labs Argo Workflows instance
+
+inputs:
+  instance:
+    description: |
+      The instance to use (`dev` or `ops`). Defaults to `ops`.
+    default: ops
+  namespace:
+    description: |
+      Required. The namespace to trigger the workflow in.
+    required: true
+  command:
+    description: |
+      The command to run. Defaults to `submit`.
+    default: submit
+  parameters:
+    description: |
+      The newline-separated parameters to pass to the Argo workflow. Ex:
+
+      ```
+      param1=value1
+      param2=value2
+      ```
+  workflow_template:
+    description: |
+      The workflow template to use. Defaults to `grafana-ci`. Required if `command` is `submit`.
+  extra_args:
+    description: |
+      Extra arguments to pass to the Argo CLI. Ex: `--generate-name foo-"
+  log_level:
+    description: |
+      The log level to use. Choose from `debug`, `info`, `warn` or `error`. Defaults to `info`.
+    default: info
+
+outputs:
+  uri:
+    description: |
+      The URI of the workflow that was triggered.
+    value: ${{ steps.run.outputs.uri }}
+
+runs:
+  using: composite
+
+  steps:
+    - name: Checkout
+      env:
+        action_repo: ${{ github.action_repository }}
+        action_ref: ${{ github.action_ref }}
+      uses: actions/checkout@v4.1.1
+      with:
+        repository: ${{ env.action_repo }}
+        ref: ${{ env.action_ref }}
+
+    - name: Get Argo Token
+      id: get-argo-token
+      uses: "./actions/get-vault-secrets"
+      with:
+        vault_instance: ${{ inputs.instance }}
+        repo_secrets: |
+          ARGO_TOKEN=argo-workflows-trigger:token
+
+    - name: Restore cache
+      id: restore
+      uses: actions/cache/restore@v3.3.2
+      with:
+        path: ${{ github.workspace }}/bin
+        key: argo-linux-amd64-3.5.1
+
+    - name: Fetch Github Release Asset
+      id: fetch_asset
+      uses: dsaltares/fetch-gh-release-asset@1.1.1
+      with:
+        repo: "argoproj/argo-workflows"
+        version: "tags/v3.5.1"
```

Comment:
I wonder if we should make the version an actual variable since it's referenced in line 68 as well, in case we want to update it and we don't want to miss a thing.

Reply:
Okay, you can't have real variables like that, but you can do a similar thing with "setup" step that sets an env var, let me do that.

Reply:
Works, I've done it in 001829f6a0f4cfde04b1e4e580b38338b8d380cf and you can see a run [here](https://github.com/grafana/shared-workflows/actions/runs/7208163232/job/19636503883). Don't worry that it failed - that's because I undeployed the resources this requires to work. We'll bring those back soon. Just notice that the binary was fetched properly 👍